packages/blocks/src/api/parser/convert-legacy-block.js

```diff
@@ -77,6 +77,35 @@ export function convertLegacyBlockNameAndAttributes( name, attributes ) {
 		newAttributes.legacy = true;
 	}
 
+	// Grid layout attributes were stored as strings in WP 6.6. Convert them to numbers.
+	if (
+		attributes.layout?.type === 'grid' &&
+		typeof attributes.layout?.columnCount === 'string'
+	) {
+		newAttributes.layout = {
+			...newAttributes.layout,
+			columnCount: parseInt( attributes.layout.columnCount, 10 ),
+		};
+	}
+	if ( typeof attributes.style?.layout?.columnSpan === 'string' ) {
+		newAttributes.style = {
+			...newAttributes.style,
+			layout: {
+				...newAttributes.style.layout,
+				columnSpan: parseInt( attributes.style.layout.columnSpan, 10 ),
```

Comment:
Should we check the string isn't empty here? Both `columnSpan` and `rowSpan` can be empty strings if the input has been cleared. It shouldn't make much difference in practice but it does pop up a console warning.

Reply:
Ah good point. I'll add a `isNaN` check in case it's an empty string or not a number.